ci/jenkinsfiles/common-lib.groovy

```diff
@@ -0,0 +1,247 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Contributors:
+*     Abdoul BA <aba@nuxeo.com>
+*/
+
+void compile() {
+  echo '''
+    ----------------------------------------
+    Compile
+    ----------------------------------------
+  '''
+  sh "mvn ${MAVEN_ARGS} -V -T0.8C -DskipTests clean install"
+}
+
+String getVersion() {
+  return this.isPullRequest() ? this.getPullRequestVersion() : this.getReleaseVersion()
+}
+
+String getReleaseVersion() {
+  String preid = 'rc'
+  String nextPromotion = readMavenPom().getVersion().replace('-SNAPSHOT', '')
+  String version = "${nextPromotion}-${preid}.0" // first version ever
+
+  // find the latest tag if any
+  sh "git fetch origin 'refs/tags/v${nextPromotion}*:refs/tags/v${nextPromotion}*'"
+  String tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${nextPromotion}*' | tail -1 | tr -d '\n'")
+  if (tag) {
+    container('maven') {
+      version = sh(returnStdout: true, script: "npx --ignore-existing semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+String getPullRequestVersion() {
+  return "${readMavenPom().getVersion()}-${BRANCH_NAME}"
+}
+
+void gitCheckout(String reference) {
+  sh "git checkout ${reference}"
+}
+
+void gitCommit(String message, String option) {
+  echo '''
+    ----------------------------------------
+    Git Commit
+    ----------------------------------------
+  '''
+  sh """
+    git commit ${option} -m "${message}"
+  """
+}
+
+void gitPush(String reference) {
+  sh "git push origin ${reference}"
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+def isPullRequest() {
+  return "${BRANCH_NAME}" =~ /PR-.*/
+}
+
+void lint() {
+  echo '''
+    ----------------------------------------
+    Run Linting Validations
+    ----------------------------------------
+  '''
+  sh """
+    cd ${FRONTEND_FOLDER}
+    npm run lint
+  """
+}
+
+String mavenArgs() {
+  String args = '-DskipITs=true -fae -B -nsu'
+  if (!this.isPullRequest()) {
+    args += ' -Prelease'
+  }
+  return args
+}
+
+boolean needsSaucelabs() {
+  return this.isPullRequest() && pullRequest.labels.contains('saucelabs')
+}
+
+void runFrontEndUnitTest() {
+  return {
+    stage('Run Unit tests: FrontEnd') {
+      container('maven') {
+        String context = 'coldstorage/utests/frontend'
+        String message = 'Unit tests - FrontEnd'
+        String strCommand = """
+          cd ${FRONTEND_FOLDER}
+          npm clean-install
```

Comment:
Not sure if we need/want to have this overhead here.
We are "redefining" what some maven phases are doing (see [here](https://github.com/nuxeo/nuxeo-coldstorage/blob/master/nuxeo-coldstorage-web/pom.xml#L16-L39))

And the `build.xml` [file](https://github.com/nuxeo/nuxeo-coldstorage/blob/master/nuxeo-coldstorage-web/build.xml) has the mappings.

This to say that when we reach this point (since we've done already the `mvn clean install` we'll have already the npm dependencies). Anyway, if we foresee the need to get them again, it may worth to rethink the flow to avoid fetching everything more than once.
WDYT?

Reply:
we can remove the maven install step from compile.
and run these 3 steps ( install , lint, and unit tests) in the playwright container.